src/main/php/com/handlebarsjs/HandlebarsParser.class.php

```diff
@@ -193,11 +216,96 @@ protected function initialize() {
   }
 
   /**
-   * Returns parsing target
+   * Parse a template
    *
-   * @return com.github.mustache.NodeList
+   * @param  string $template The template as a string
+   * @param  string $start Initial start tag, defaults to "{{"
+   * @param  string $end Initial end tag, defaults to "}}"
+   * @param  string $indent What to prefix before each line
+   * @return com.github.mustache.Node The parsed template
+   * @throws com.github.mustache.TemplateFormatException
    */
-  protected function target() {
-    return new Nodes();
+  public function parse(Tokenizer $tokens, $start= '{{', $end= '}}', $indent= '') {
+    $state= new ParseState();
+    $state->target= new Nodes();
+    $state->start= $start;
+    $state->end= $end;
+    $state->parents= [];
+    $standalone= implode('', array_keys($this->standalone));
+
+    // Tokenize template
+    $tokens->delimiters= "\n";
+    $tokens->returnDelims= true;
+    while ($tokens->hasMoreTokens()) {
+      $token= $tokens->nextToken();
+
+      // Yield empty lines as separate text nodes
+      if ("\n" === $token) {
+        $state->target->add(new TextNode($indent.$token));
+        continue;
+      }
+
+      $line= $indent.$token.$tokens->nextToken();
+      $offset= 0;
+      $length= strlen($line);
+      do {
+        $state->padding= '';
+
+        // Find first unescaped `{{` starting at $offset
```

Comment:
This block is the only part where this parser differentiates from the Mustache parser, which doesn't allow escaping tags.